lib/facter/firewalld_state.rb

```diff
@@ -0,0 +1,14 @@
+def state
+  Facter::Util::Resolution.exec('firewall-cmd --state 2> /dev/null')
+end
+
+Facter.add(:firewalld_state) do
+  confine :kernel => :linux
+  setcode do
+    if state == "running"
+      firewalld_state = "true"
+    else 
+      nil
+    end
+  end
+end
```

Comment:
Here's an example of a unit test for a fact

https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/spec/unit/facter/root_home_spec.rb and the fact at https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/lib/facter/root_home.rb